### PR TITLE
Implement and document `hide-` classes

### PR DESCRIPTION
**Pull request by DerFlatulator**
As requested in #138, I've implemented and documented some classes that provide the ability to hide or show content based on viewport size.

The classes are:

- `hide-xs`, `hide-sm`, `hide-md`, `hide-lg`: Hide only on the viewports specified.
- `hide-sm-up`: Same as `hide-sm hide-md hide-lg`.
- `hide-md-up`: Same as `hide-md hide-lg`.
- `hide-sm-down`: Same as `hide-xs hide-sm`.
- `hide-md-down`: Same as `hide-xs hide-sm hide-md`.

New documentation visible at: http://derflatulator.github.io/flexboxgrid/#hiding 

There is no real need for `show-` classes, as at most two `hide-` classes would be required for any configuration.

### CODE REVIEW

File: dist/flexboxgrid.css [hunk 1]
```diff
@@ -1043,4 +1051,38 @@
...
+  }
+}
+
+@media only screen and (max-width: 64em) {
+  .hide-sm-down {
+    display: none !important;
+  }
+}
+
+@media only screen and (max-width: 75em) {
```
Comment (visnup) [inline dist/flexboxgrid.css]: all of these `max-width`s should probably be more like 74.9375em otherwise they apply too far up?

Comment (DerFlatulator) [inline dist/flexboxgrid.css]: Possibly. How did you determine that number exactly?

**Comment by kristoferjoseph** [comment]
Flexbox grid will not be adding this feature.

**Comment by boustanihani** [comment]
@kristoferjoseph could you please explain why :disappointed_relieved: ? IMO this is very useful...

**Comment by kristoferjoseph** [comment]
screenshot 2016-01-28 11 26 39

**Comment by bitdivision** [comment]
Quoting Ethan Marcotte: 

> 'But that said, there're definitely *edge* cases where it's handy. I just try to avoid using it as a default approach, personally.'

As far as I can tell, most of the work for this has been done, there's seemingly little to lose by adding it to the project (other than a small amount of bloat), and it does handily solve those edge cases mentioned above.

**Comment by boustanihani** [comment]
@bitdivision :+1:

**Comment by alexhawkins** [comment]
+1 on adding

**Comment by tzapu** [comment]
+1 on adding ...
if you ve got a column you really don t want on mobiles, maybe some navigation shortcuts, or banners or whatever, it would be quite handy to hide them...

**Comment by smooth-operator** [comment]
+1 on adding

**Comment by hufeihxw** [comment]
Isn't hiding considered as some kind of "different behavior" too?
Another edging case is server side rendering, server side codes will most likely return the complete dom structure without considering media-query

**Comment by amalahussein** [comment]
+1 on adding this... this is a super helpful feature.

**Comment by djensen47** [comment]
One use case for this is advertising. You don't want to try to show a 300x250 add block on a mobile screen. Instead, you want to use the mobile add. Responsive ads are getting better but it is still very nice to avoid showing big ad blocks.

**Comment by djensen47** [comment]
Looks like there is this now: https://github.com/lgg/flexboxgrid-helpers

Maybe add a link to the docs?

**Comment by kristoferjoseph** [comment]
I think it will make it into 7.0

**Pull request closed without merging**